ui/raidboss/browser_tts_engine.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const TTSEngineType = {
+  SpeechSynthesis: 0,
+  GoogleTTS: 1,
+};
+
+class TTSItem {
+  play() {}
+}
+
+class SpeechTTSItem extends TTSItem {
+  constructor(text) {
+    super();
+    this.text = text;
+    this.item = new SpeechSynthesisUtterance(text);
+  }
+
+  play() {
+    window.speechSynthesis.speak(this.item);
+  }
+}
+
+class GoogleTTSItem extends TTSItem {
+  constructor(text) {
+    super();
+    this.text = text;
+    let iframe = document.createElement('iframe');
+    // remove sandbox so we can modify contents/call play on audio element later
+    iframe.removeAttribute('sandbox');
+    iframe.style.display = 'none';
+    document.body.appendChild(iframe);
+    let encText = encodeURIComponent(text);
+    iframe.contentDocument.body.innerHTML = '<audio src="https://translate.google.com/translate_tts?ie=UTF-8&client=tw-ob&tl=en&q=' + encText + '" id="TTS">';
+    this.item = iframe.contentDocument.body.firstElementChild;
+  }
+
+  play() {
+    this.item.play();
+  }
+}
+
+class BrowserTTSEngine {
+  constructor() {
+    engineType = null;
+    ttsItems = {};
+  }
+
+  play(text) {
+    try {
+      if (this.ttsItems[text] !== undefined) {
+        this.ttsItems[text].play();
+      } else {
+        switch (this.engineType) {
+        case null:
+          // figure out what TTS engine type we need
+          if (window.speechSynthesis !== undefined &&
+            window.speechSynthesis.getVoices().length > 0)
+            this.engineType = TTSEngineType.SpeechSynthesis;
```

Comment:
Ok, sorry.  I led you astray, I think.  I didn't realize that `speechSynthesis` got caught up in the Chrome autoplay shenanigans and you can't use it before a user interacts with the document, so it's very likely unusable.  Maybe better to just remove the complexity.  I was hoping it would be easier to just lean on an existing web api instead of having to depend on a service (which is more likely to change or have errors, etc), but apparently that's not possible.

Sorry to waste your time with a half-baked suggestion.

Reply:
They both suffer from the same problem vs autoplay. The reason for the fallback logic and external service is to address if the API is unavailable for some reason or if there are no voices available (which is what I've seen with my own mobile device and chromium/chrome/vivaldi on linux).

Reply:
Oh right, of course it would be the same autoplay issue.  I shouldn't do code review while exhausted.  This is all fine, then.

Ok! Sorry to push on this.  Maybe I had just hoped there could be an easier solution somewhere here.

Reply:
The best I could think of is to detect the autoplay issue (there's google documentation on how to detect that interaction is still required) and prompt the user to click a button or something if we're using browser tts instead of ACT tts.

Reply:
Yeah, that sounds like a good approach.  I wonder if the button should be at load time though, as it would not be great to be midfight and suddenly have a button to hit before tts works.

Reply:
Maybe add the button (as a position absolute to avoid layout issues) in the `BrowserTTSEngine` constructor if interaction is still required? That should get triggered on page load.

Reply:
Yeah, I'm not sure what makes the most sense.  Maybe adding a button would be a good followup patch and I can merge this without it to start so other people can start using it too.

Reply:
I can code this up tomorrow sometime. Would you want the styles for this put in a stylesheet somewhere or should I just do the entire button inline?

Reply:
Probably the normal raidboss.css file is a good spot.